tensorboard/compat/tensorflow_stub/io/gfile.py

```diff
@@ -395,8 +483,37 @@ def __next__(self):
     def next(self):
         return self.__next__()
 
+    def flush(self):
+        if self.write_temp is not None:
+            # read temp file from the beginning
+            self.write_temp.flush()
+            self.write_temp.seek(0)
+            chunk = self.write_temp.read()
+            if chunk is not None:
+                # use append if filesystem supports it
+                if self.fs_supports_append:
+                    if not self.write_started:
+                        # write the first chunk
+                        self.fs.write(self.filename, chunk, self.binary_mode)
+                        self.write_started = True
+                    else:
+                        # append the later chunks
+                        self.fs.append(self.filename, chunk, self.binary_mode)
+                    # remove temp file, since we are appending
+                    self.write_temp.close()
+                    self.write_temp = None
+                else:
+                    # write full contents and keep in temp file
+                    self.fs.write(self.filename, chunk, self.binary_mode)
+                    self.write_temp.seek(len(chunk))
+
     def close(self):
-        pass
+        self.flush()
+        if  self.write_temp is not None:
+            self.write_temp.close()
+            self.write_temp = None
+            self.write_started = False
+        self.closed = True
```

Comment:
We never read the value of `closed` - we probably should to e.g. to block a write() or flush() on a closed file.